src/libtls/tls_server.c

```diff
@@ -275,10 +305,10 @@ static bool retrying(private_tls_server_t *this)
 static status_t process_client_hello(private_tls_server_t *this,
 									 bio_reader_t *reader)
 {
-	uint16_t legacy_version = 0, version = 0, key_share_length, key_type = 0;
-	uint16_t extension_type = 0;
+	uint16_t legacy_version = 0, version = 0, extension_type = 0;
 	chunk_t random, session, ciphers, versions = chunk_empty, compression;
-	chunk_t ext = chunk_empty, key_share = chunk_empty;
+	chunk_t ext = chunk_empty, key_shares = chunk_empty;
+	key_share_t peer;
```

Comment:
Hm, probably I have to initialise `peer.key_share` with a `chunk_empty` here to have a properly initialised `peer.key_share.len`, right?

Reply:
Not really, since you only use it when `peer_offered_curve()` returned TRUE, in which case both members are fully defined, and a DH object was created.